src/service/action-impl.js

```diff
@@ -87,17 +87,17 @@ export class ActionInvocation {
 export class ActionService {
 
   /**
-   * @param {!Window} win
+   * @param {!./ampdoc-impl.AmpDoc} ampdoc
    */
-  constructor(win) {
-    /** @const {!Window} */
-    this.win = win;
+  constructor(ampdoc) {
+    /** @const {!./ampdoc-impl.AmpDoc} */
+    this.ampdoc = ampdoc;
 
     /** @const @private {!Object<string, function(!ActionInvocation)>} */
     this.globalMethodHandlers_ = {};
 
     /** @private {!./vsync-impl.Vsync} */
-    this.vsync_ = vsyncFor(this.win);
+    this.vsync_ = vsyncFor(ampdoc.getWin());
```

Comment:
So, for the time being, at least until most of migration is done, I'd really like to make a clear line between a window service and a ampdoc service, so I'd like to keep APIs as specific as possible w.r.t. arguments.